code/game/objects/items/sharpener.dm

```diff
@@ -24,18 +24,19 @@
 	if(istype(I, /obj/item/melee/transforming/energy))
 		to_chat(user, "<span class='warning'>You don't think \the [I] will be the thing getting modified if you use it on \the [src]!</span>")
 		return
-	if(istype(I, /obj/item/twohanded))//some twohanded items should still be sharpenable, but handle force differently. therefore i need this stuff
-		var/obj/item/twohanded/TH = I
-		if(TH.force_wielded >= max)
-			to_chat(user, "<span class='warning'>[TH] is much too powerful to sharpen further!</span>")
-			return
-		if(TH.wielded)
-			to_chat(user, "<span class='warning'>[TH] must be unwielded before it can be sharpened!</span>")
+
+	// Some two handed items should still be sharpenable, but they handle force differently.
+	// Therefore this stuff is needed
+	var/datum/component/two_handed/comp_twohand = I.GetComponent(/datum/component/two_handed)
```

Comment:
this is better handled through a new signal that you could call COMSIG_SHARPEN_ITEM or something similar (Check signal.dm for ideas of how you can best name them I always forget the conventions)

Reply:
Who even did shapener can manage the signals it wants for this. It is not related to this component.

Reply:
you're refactoring 2handers. this is 2 handed behavior.

Reply:
this is behavior of more then two handed items, this sharpener code is bad but i am not reworking sharpener on this PR for it to have signals and handling by all objects in this PR.

Reply:
I'll defer to @ninjanomnom but I'm pretty sure he will agree with me; this code was alright when the item was two-handed but components listen for signals as thats their intent

Reply:
A signal for something being sharpened would be reasonable and should be done instead of the GetComponent

Reply:
This has been reworked to use signals, check this over please.